src/ripple/protocol/Quality.h

```diff
@@ -276,6 +276,40 @@ class Quality
         os << quality.m_value;
         return os;
     }
+
+    // return the relative distance (relative error) between two qualities. This is used for testing only.
+    // relative distance is abs(a-b)/min(a,b)
+    friend double
+    relativeDistance(Quality const& q1, Quality const& q2)
+    {
+        assert(q1.m_value > 0 && q2.m_value > 0);
+
+        if (q1.m_value == q2.m_value)  // make expected common case fast
+            return 0;
+
+        auto const minV = std::min(q1.m_value, q2.m_value);
+        auto const maxV = std::max(q1.m_value, q2.m_value);
+
+        auto mantissa = [](std::uint64_t rate) {
```

Comment:
What is going on here? `Quality` is a "custom floating point representation" according to its comment, but the schema is not given. Were the implementations of these `mantissa` and `exponent` functions reverse-engineered from the implementation of `round`? What if we made `mantissa` and `exponent` methods of `Quality` (they can even be private), and called them from here and `round`?

Separately, documenting the schema is a perfect opportunity to use ASCII art in a comment.

Reply:
Agreed, the "custom floating point representation" isn't good documentation. We may as well try to address this in this PR. Will do.

Reply:
Added a comment `a1a67d6d0 Add fix amendment`. It's the same format as STAmount.